src/prefect/client/client.py

```diff
@@ -188,35 +190,37 @@ def __init__(
 
     # API key authentication -----------------------------------------------------------
 
-    def get_auth_tenant(self) -> str:
+    def _get_auth_tenant(self) -> str:
         """
         Get the current tenant associated with the API key being used. If the client has
         a specific tenant id set, this will verify that the given tenant id is
```

Comment:
how does this method actually do the verification? it because we're passing "x-prefect-tenant-id" as a header?
I think this comment has already tripped me up once before, maybe worth adding to the docstring

Reply:
Since we're attaching the tenant id the backend raise an unauthenticated error if it's not a valid tenant for the key. I've added a bit more to the comment.